### PR TITLE
CI pipeline implemented

### PR DESCRIPTION
**Pull request by adityadangeska**
- CI pipeline implemented
- Docker image is built using SKA docker image
- test cases are running

### CODE REVIEW

**Review by ajoubertza — COMMENTED** [review]
This looks good (high-level review, not looking at all the details)
Do you still plan to get the build results linked/pushed back into github?  It would be great to see them against each commit in future.

**Comment by adityadangeska** [comment]
> This looks good (high-level review, not looking at all the details)
> Do you still plan to get the build results linked/pushed back into github? It would be great to see them against each commit in future.

I didnt plan for this. But I like your suggestion. I think it will be very useful to see the result of each of the commit. I will plan for this in the upcoming PI.

File: .gitlab-ci.yml [hunk 1]
```diff
@@ -0,0 +1,42 @@
...
+before_script:
+  - docker login -u $DOCKER_REGISTRY_USER_LOGIN -p $CI_REGISTRY_PASS_LOGIN $CI_REGISTRY
+
+
+# The YAML file defines a set of jobs with constraints stating when they should be run.
+# You can specify an unlimited number of jobs which are defined as top-level elements with an arbitrary name and always have to contain at least the script clause.
+# In this case we have only the test job which produce an artifacts (it must be placed into a directory called "public")
+# It is also specified that only the master branch will be subject of this job.
+
+stages:
```
Comment (jyotinpo) [inline .gitlab-ci.yml]: I think it is better if we also publish the results as "pages" on SKA CI dashboard at this link:
https://ska-telescope.gitlab.io/ska_ci_dashboard/

In order to do that, we can take reference from DishMaster project and align the .gitlab.yml file accordingly. I am in the middle of applying this modification.

File: Jenkinsfile [hunk 2]
```diff
@@ -1,66 +0,0 @@
-node('docker') {
-
-    withDockerContainer(
-        image: 'tango-lmc-base-classes:latest',
```
Comment (jyotinpo) [inline Jenkinsfile]: This Jenkins file can now be safely deleted.

File: code-analysis.sh [hunk 3]
```diff
@@ -0,0 +1,19 @@
...
+echo "===================="
+echo
+
+echo "MODULE ANALYSIS"
+echo "---------------"
+pylint --rcfile=rcfile skabase
+
+echo "TESTS ANALYSIS"
+echo "--------------"
+pylint skabase/SKAAlarmHandler/test
```
Comment (jyotinpo) [inline code-analysis.sh]: I think we also need to include rcfile for the Test Analysis as well?

File: .gitlab-ci.yml (hunk 1 above)
Comment (jyotinpo) [reply .gitlab-ci.yml]: Tried implementing the changes, but still, the dashboard is not reflecting the results. I will check with the System team on this issue.

**Comment by jyotinpo** [comment]
- make test if getting failed on the local machine. 
- Tried implementing the changes, but still, the dashboard is not reflecting the results. I will check with the System team on this issue.

**Review by jyotinpo — APPROVED** [review]
SKA CI Dashboard now updating correctly. Checked following links:
https://ska-telescope.gitlab.io/ska_ci_dashboard/
https://ska-telescope.gitlab.io/lmc-base-classes/
The results will be reflected in term of percentage in the next scheduled run of the SKA CI Dashboard project.

**Pull request merged**